addons/account/report/account_bank_reconciliation_report.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Odoo, Open Source Management Solution
+#    Copyright (C) 2004-2014 OpenErp S.A. (<http://odoo.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class account_bank_reconciliation(models.AbstractModel):
+    _name = 'account.bank.reconciliation'
+    _description = 'Bank reconciliation report'
+
+    @api.model
+    def get_lines(self, context_id, line_id=None):
+        return self.with_context(
+            date_from=context_id.date_from,
+            journal_id=context_id.journal_id,
+            context_id=context_id,
+        )._lines()
+
+    @api.model
+    def _lines(self):
+        lines = []
+        #Start amount
+        bank_lines = self.env['account.bank.statement.line'].search([['statement_id.journal_id', '=', self.env.context['journal_id'].id],
+                                                                     ['date', '<=', self.env.context['date_from']]])
+        reconcile_lines = filter(lambda line: len(line.journal_entry_ids) > 0, bank_lines)
```

Comment:
instead of one search filtered 3 times, i'd rather see 3 different searches. It should be less greedy